pkg/catalog/manager/traverse.go

```diff
@@ -54,36 +62,42 @@ func (m *Manager) traverseAndUpdate(helm *helmlib.Helm, commit string, cmt *Cata
 		}
 	}
 
-	entriesWithErrors, entriesToProcess := m.preprocessCatalog(catalog, index.IndexFile.Entries)
-	for chart, metadata := range entriesToProcess {
-		newHelmVersionCommits[chart] = v32.VersionCommits{
-			Value: map[string]string{},
+	catalogTemplateMap, err := m.getTemplateMap(catalogName, templateNamespace)
+	if err != nil {
+		return err
+	}
+
+	// Check if a version has been deleted from a template or if an existing template version has changed.
+	hasChartChanged := func(template *v3.CatalogTemplate, metadata helmlib.ChartVersions) bool {
+		if template == nil || (len(metadata) != len(template.Spec.Versions)) {
+			return true
 		}
-		existingHelmVersionCommits := map[string]string{}
-		if catalog.Status.HelmVersionCommits[chart].Value != nil {
-			existingHelmVersionCommits = catalog.Status.HelmVersionCommits[chart].Value
+
+		metadataVersions := make(map[string]string)
+		for _, metadataVersion := range metadata {
+			metadataVersions[metadataVersion.Version] = metadataVersion.Digest
 		}
-		keywords := map[string]struct{}{}
-		// comparing version commit with the previous commit to detect if a template has been changed.
-		hasChanged := false
-		versionNumber := 0
-		for _, version := range metadata {
-			newHelmVersionCommits[chart].Value[version.Version] = version.Digest
-			digest, ok := existingHelmVersionCommits[version.Version]
-			if !ok || digest != version.Digest {
-				hasChanged = true
-			}
-			if ok {
-				versionNumber++
+		for _, templateVersion := range template.Spec.Versions {
+			// If the digest is not the same between the actual and desired version, or if the version does not
+			// exist in the metadata, then we know that the chart has changed. In addition, we do not have to check
+			// if metadata versions exist because the length of each slice is the same at this point.
```

Comment:
Couldn't a version be removed and another added leading to equal lengths: `because the length of each slice is the same at this point.`?

Reply:
Is this not solved by line 72?